client/components/payment-activity/date-range-picker.tsx

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import moment from 'moment';
+import { recordEvent } from 'wcpay/tracks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InlineLabelSelect from '../inline-label-select';
+import { DateRange } from './types';
+
+const now = moment();
+const yesterdayEndOfDay = moment()
+	.clone()
+	.subtract( 1, 'd' )
+	.set( { hour: 23, minute: 59, second: 59, millisecond: 0 } );
+const todayEndOfDay = moment()
+	.clone()
+	.set( { hour: 23, minute: 59, second: 59, millisecond: 0 } );
+const timeOptions: {
+	[ key: string ]: {
+		start: moment.Moment;
+		end: moment.Moment;
+		displayKey: string;
+	};
+} = {
+	today: {
+		start: now
+			.clone()
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: todayEndOfDay,
+		displayKey: __( 'Today', 'woocommerce-payments' ),
+	},
+	last_7_days: {
+		start: now
+			.clone()
+			.subtract( 7, 'days' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 7 days', 'woocommerce-payments' ),
+	},
+	last_4_weeks: {
+		start: now
+			.clone()
+			.subtract( 4, 'weeks' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 4 weeks', 'woocommerce-payments' ),
+	},
+	last_3_months: {
+		start: now
+			.clone()
+			.subtract( 3, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 3 months', 'woocommerce-payments' ),
+	},
+	last_12_months: {
+		start: now
+			.clone()
+			.subtract( 12, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 12 months', 'woocommerce-payments' ),
+	},
+	month_to_date: {
+		start: now.clone().startOf( 'month' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Month to date', 'woocommerce-payments' ),
+	},
+	quarter_to_date: {
+		start: now.clone().startOf( 'quarter' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Quarter to date', 'woocommerce-payments' ),
+	},
+	year_to_date: {
+		start: now.clone().startOf( 'year' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Year to date', 'woocommerce-payments' ),
+	},
+};
+
+const formatDateRange = (
+	start: moment.Moment,
+	end: moment.Moment
+): string => {
+	// Today - show only today's date.
+	if ( start.isSame( end, 'day' ) ) {
+		return start.format( 'MMMM D, YYYY' );
+	}
+
+	// Different years - show year for both start and end
+	if ( ! start.isSame( end, 'year' ) ) {
+		return `${ start.format( 'MMMM D, YYYY' ) } - ${ end.format(
+			'MMMM D, YYYY'
+		) }`;
+	}
+
+	// Same year - show year only for end date.
```

Comment:
I did consider this, but decided to go with[ the design](https://www.figma.com/design/bIfbBgyBPRzvXEl7dEg3cv/Payment-Widget?node-id=1801-115180&m=dev) that shows the months even if they are same. This may be done probably to ensure alignment, but allow me to check with @rogermattic .

Reply:
I did a quick check and see that it looks more misaligned if we remove month from the starting date for same month 

#### With changes
<img width="400" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/63206168-a7b2-4536-bbd3-13fe9575e7d9">

#### As it is currently 
<img width="380" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/0d80a641-f944-44a1-97ee-4fcaede3501b">

Reply:
I think it's reasonable to show e.g. `June 5 - June 11` - it's clearer. I am assuming this is intended by design as well. 

`5 – June 11, 2024` is unusual formatting IMO, not worth the few characters saved.

Reply:
Feel free to log an issue for that @rogermattic - relatively easy fix :)